### PR TITLE
Only log vagrant output on failure

### PR DESCRIPTION
**Pull request by alpar-t**
This PR adds a prefix consisting of the box name to the console output.
This makes it easy to diferentiate the Gradle output comming from inside
the VM.

The Standard error is sent to a file that is included in the gcp upload.
This is the vagrant debug log that used to go to the console too.

There's also a switch from  `LoggedExec`, that was probably an owersight
as it had no effect as we were printing on the console anyhow as part of
the progress logger.

### CODE REVIEW

File: buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/VagrantMachine.java [hunk 1]
```diff
@@ -69,7 +73,7 @@ public void execute(Action<VagrantExecSpec> action) {
 
         Objects.requireNonNull(vagrantSpec.command);
 
-        LoggedExec.exec(project, execSpec -> {
+        project.exec(execSpec -> {
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/VagrantMachine.java]: The intention of LoggedExec was to not output the potentially very large output for the task if it succeeds. I'm not sure we want to lose that.

File: buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/VagrantMachine.java [hunk 2]
```diff
@@ -198,7 +210,7 @@ protected void logLine(String line) {
             if (progress != null) {
                 progressLogger.progress(progress);
             }
-            System.out.println(line);
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/VagrantMachine.java]: This was actually leftover from debugging. I don't think we want this always going to stdout. In fact, this ProgressOutputStream is meant only to write to the progress logger.

File: buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/VagrantMachine.java [hunk 3]
```diff
@@ -87,14 +91,20 @@ public void execute(Action<VagrantExecSpec> action) {
...
             }
-            OutputStream output = execSpec.getStandardOutput();
-            // output from vagrant needs to be manually curated because --machine-readable isn't actually "readable"
-            OutputStream progressStream = new ProgressOutputStream(vagrantSpec.command, progressHandler);
-            execSpec.setStandardOutput(new TeeOutputStream(output, progressStream));
+
+            execSpec.setStandardOutput(new ProgressOutputStream(vagrantSpec.command, progressHandler, System.out));
+            try {
+                execSpec.setErrorOutput(
+                    Files.newOutputStream(project.getBuildDir().toPath().resolve("vagrant.log"), CREATE, APPEND)
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/VagrantMachine.java]: LoggedExec has an option to spool the output to disk (although we may need to convert the static exec method to support that option).

File: buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/VagrantMachine.java [hunk 4]
```diff
@@ -87,14 +91,20 @@ public void execute(Action<VagrantExecSpec> action) {
...
 
+            // output from vagrant needs to be manually curated because --machine-readable isn't actually "readable"
             UnaryOperator<String> progressHandler = vagrantSpec.progressHandler;
             if (progressHandler == null) {
                 progressHandler = new VagrantProgressLogger("==> " + extension.getBox() + ": ");
             }
-            OutputStream output = execSpec.getStandardOutput();
-            // output from vagrant needs to be manually curated because --machine-readable isn't actually "readable"
-            OutputStream progressStream = new ProgressOutputStream(vagrantSpec.command, progressHandler);
-            execSpec.setStandardOutput(new TeeOutputStream(output, progressStream));
```
Comment (rjernst) [inline buildSrc/src/main/java/org/elasticsearch/gradle/vagrant/VagrantMachine.java]: I think the tee was fine? We just need a wrapper around the tee output stream to prefix each line. We used to do this and lost it somewhere.

**Comment by atorok** [comment]
@rjernst that `System.out` threw me off and took me in the other direction. I updated the pr to remove it. I think we are ok without that prefix since we'll only have the output from one invocation

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by atorok** [comment]
@elasticmachine update branch

**Comment by atorok** [comment]
@elasticmachine update branch

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/packaging-matrix

**Comment by atorok** [comment]
@elasticmachine update branch

**Pull request merged**